### PR TITLE
Module Revamp(Alpha) Add Java Support to Basic Complete Search

### PR DESCRIPTION
**Pull request by brad182**
Part of the module revamp. Is the contributors thing going to be added soon because I've only provided solutions and not major changes

### CODE REVIEW

**Comment by maggie-j-liu** [comment]
I'll just add you as contributor in my pr

File: content/2_Bronze/Intro_Complete.mdx [hunk 1]
```diff
@@ -61,23 +61,44 @@ int main() {
...
+		for (int i = 0; i < n; i++) {
+			y[i] = io.nextInt();
+		}
+
+		int max = 0; // storing the current maximum
+		for(int i = 0; i < n; i++){ // for each first point
+			for(int j = i+1; j < n; j++){ // for each second point
+				int dx = x[i] - x[j];
+				int dy = y[i] - y[j];
+				max = Math.max(max, dx*dx + dy*dy);
```
Comment (maggie-j-liu) [inline content/2_Bronze/Intro_Complete.mdx:89]:
```suggestion
			for(int j = i + 1; j < n; j++){ // for each second point
				int dx = x[i] - x[j];
				int dy = y[i] - y[j];
				max = Math.max(max, dx * dx + dy * dy);
```

File: content/2_Bronze/Intro_Complete.mdx [hunk 2]
```diff
@@ -167,6 +188,63 @@ Rounding suffices (`(int)round(pow(max_dist,2))`), but the takeaway is that you
...
+However, it still fails on the following test case (it outputs 12, but the
+answer is 13):
+
+```
+2
+0 3
+2 0
+```
+
+Rounding suffices (`(int)round(pow(max_dist,2))`), but the takeaway is that you
```
Comment (maggie-j-liu) [inline content/2_Bronze/Intro_Complete.mdx:243]: can you change this to match your java code (and is this true for your code?)

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: content/2_Bronze/Intro_Complete.mdx [hunk 3]
```diff
@@ -61,23 +61,44 @@ int main() {
...
+
+		for (int i = 0; i < n; i++) {
+			x[i] = io.nextInt();
+		}
+		for (int i = 0; i < n; i++) {
+			y[i] = io.nextInt();
+		}
+
+		int max = 0; // storing the current maximum
+		for(int i = 0; i < n; i++){ // for each first point
```
Comment (nwatx) [inline content/2_Bronze/Intro_Complete.mdx:85]:
```suggestion
		for(int i = 0; i < n; i++) { // for each first point
```

File: content/2_Bronze/Intro_Complete.mdx [hunk 4]
```diff
@@ -61,23 +61,44 @@ int main() {
...
+			for(int j = i + 1; j < n; j++){ // for each second point
```
Comment (nwatx) [inline content/2_Bronze/Intro_Complete.mdx:86]:
```suggestion
			for(int j = i + 1; j < n; j++) { // for each second point
```

**Review by nwatx — CHANGES_REQUESTED** [review]

File: content/2_Bronze/Intro_Complete.mdx [hunk 5]
```diff
@@ -61,23 +61,44 @@ int main() {
...
+				int dx = x[i] - x[j];
+				int dy = y[i] - y[j];
+				max = Math.max(max, dx * dx + dy * dy);
+				// if the square of the distance between the two points is greater than
+				// our current maximum, then update the maximum
```
Comment (SansPapyrus683) [inline content/2_Bronze/Intro_Complete.mdx:91]: put this above that line with the `max`

File: content/2_Bronze/Intro_Complete.mdx [hunk 6]
```diff
@@ -167,6 +188,63 @@ Rounding suffices (`(int)round(pow(max_dist,2))`), but the takeaway is that you
...
+
+		for (int i = 0; i < n; i++) {
+			x[i] = io.nextInt();
+		}
+		for (int i = 0; i < n; i++) {
+			y[i] = io.nextInt();
+		}
+
+		double maxSquare = 0; // storing the current maximum
+		for(int i = 0; i < n; i++){ // for each first point
```
Comment (SansPapyrus683) [inline content/2_Bronze/Intro_Complete.mdx:217]:
```suggestion
		for (int i = 0; i < n; i++) {  // for each first point
```
and the for loop below as well

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
small formatting things

**Review by SansPapyrus683 — APPROVED** [review]

File: content/2_Bronze/Intro_Complete.mdx [hunk 7]
```diff
@@ -167,6 +188,63 @@ Rounding suffices (`(int)round(pow(max_dist,2))`), but the takeaway is that you
...
+			}
+		}
+		io.println((int) Math.pow(maxSquare, 2));
+		io.close();
+	}
+}
+
+```
+
+However, it still fails on the following test case (it outputs 12, but the
```
Comment (nwatx) [inline content/2_Bronze/Intro_Complete.mdx:234]: I would like it if the fact that it was incorrect was _more_ emphasized (probably bolded or in a spoiler)

File: content/2_Bronze/Intro_Complete.mdx [hunk 8]
```diff
@@ -167,6 +189,64 @@ Rounding suffices (`(int)round(pow(max_dist,2))`), but the takeaway is that you
...
+answer is 13):
+</Warning>
+	
+```
+2
+0 3
+2 0
+```
+
+Rounding suffices (`(int) Math.pow(maxSquare, 2)`), but the takeaway is that you
```
Comment (nwatx) [inline content/2_Bronze/Intro_Complete.mdx:245]: shouldn't this be math.round

Comment (HarryWangATX) [inline content/2_Bronze/Intro_Complete.mdx:245]:
```suggestion
Rounding suffices (`(int) Math.round(Math.pow(maxSquare, 2))`), but the takeaway is that you
```

**Pull request merged**